Stratis.Bitcoin.Features.Wallet/WalletManager.cs

```diff
@@ -611,10 +611,9 @@ public IEnumerable<UnspentOutputReference> GetSpendableTransactionsInWallet(stri
             this.logger.LogTrace("({0}:'{1}',{2}:{3})", nameof(walletName), walletName, nameof(confirmations), confirmations);
 
             Wallet wallet = this.GetWalletByName(walletName);
-            IEnumerable<UnspentOutputReference> res;
             lock (this.lockObject)
             {
-                res = wallet.GetAllSpendableTransactions(this.coinType, this.chain.Tip.Height, confirmations).ToArray();
+                UnspentOutputReference[] res = wallet.GetAllSpendableTransactions(this.coinType, this.chain.Tip.Height, confirmations).ToArray();
```

Comment:
In this case `res` is already enumerated using the count outside of the loop will is completely safe.
outside of the lock even if the wallet changes res will not be effected.

Reply:
yes, here again it seems like GetAllSpendableTransactions should be named GetAllSpendableTransactionsLocked and there should be extra comment that the lock must be held

and again, it seems like this could be written as 

```
            UnspentOutputReference[] res;
            lock (this.lockObject)
            {
                res = wallet.GetAllSpendableTransactions(this.coinType, this.chain.Tip.Height, confirmations).ToArray();
            }

            this.logger.LogTrace("(-):*.{0}={1}", nameof(res.Length), res.Length);
            return res;
```

so the CS would be shorter

Reply:
Will do so. Btw: this is the result of our failure of backpedalling from deferred execution :)

Reply:
could you pls explain what you mean here? i mean what do you suggest in general and what kind of failure this is and how it would be prevented